x-pack/plugins/ml/server/lib/register_settings.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { CoreSetup } from 'kibana/server';
+import { i18n } from '@kbn/i18n';
+import { schema } from '@kbn/config-schema';
+import { FILE_DATA_VISUALIZER_MAX_FILE_SIZE } from '../../common/constants/settings';
+import { MAX_FILE_SIZE } from '../../common/constants/file_datavisualizer';
+
+export function registerKibanaSettings(coreSetup: CoreSetup) {
+  coreSetup.uiSettings.register({
+    [FILE_DATA_VISUALIZER_MAX_FILE_SIZE]: {
+      name: i18n.translate('xpack.ml.maxFileSizeSettingsName', {
+        defaultMessage: 'File Data Visualizer maximum file upload size',
+      }),
+      value: MAX_FILE_SIZE,
+      description: i18n.translate('xpack.ml.maxFileSizeSettingsDescription', {
+        defaultMessage:
+          'Sets the file size limit when importing data in the File Data Visualizer. The highest supported value for this setting is 1GB.',
```

Comment:
Ditto re "File Data Visualizer":
```suggestion
          'Sets the file size limit when importing data in the Data Visualizer. The highest supported value for this setting is 1GB.',
```